include/circt/Dialect/HW/HWTypesImpl.td

```diff
@@ -138,6 +138,31 @@ def StructTypeImpl : HWType<"Struct"> {
   }];
 }
 
+// An enum type. Declares the hw::EnumType in C++.
+def EnumTypeImpl : HWType<"Enum"> {
+  let summary = "HW Enum type";
+  let description = [{
+    Represents an enumeration of values. Enums are interpreted as integers with
+    a synthesis-defined encoding.
+    !hw.enum<field1, field2>
```

Comment:
Should enums have an optional base type (sv.6.19)?

Reply:
see below